PWGMM/Tasks/analyse-mft-tracks.cxx

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+
+#include "TDatabasePDG.h"
+#include "Common/Core/MC.h"
+
+using namespace o2;
+using namespace o2::framework;
+
+using Particles = aod::McParticles;
+
+//First approach to analysing an AO2D.root file
+//written thanks to https://aliceo2group.github.io/analysis-framework/docs/tutorials/analysistask.html
+
+struct analyseMFTTracks {
+  int icoll = 0;
+
+  HistogramRegistry registry{
+    "registry",
+    {
+      {"TracksPhiEta_in_coll", "; #varphi; #eta; tracks", {HistType::kTH2F, {{600, -M_PI, M_PI}, {35, -4.5, -1.}}}},
+      {"TracksPhiEta", "; #varphi; #eta; tracks", {HistType::kTH2F, {{600, -M_PI, M_PI}, {35, -4.5, -1.}}}},                                                           //
+      {"TracksPhiEtaGen", "; #varphi; #eta; tracks", {HistType::kTH2F, {{600, 0, 2 * M_PI}, {35, -4.5, -1.}}}},                                                        //
+      {"TracksEtaZvtx", "; #eta; Z_{vtx}; tracks", {HistType::kTH2F, {{35, -4.5, -1.}, {201, -20.1, 20.1}}}},                                                          //
+      {"NtrkZvtx", "; N_{trk}; Z_{vtx}; events", {HistType::kTH2F, {{301, -0.5, 300.5}, {201, -20.1, 20.1}}}},                                                         //
+      {"NtrkEta", "; N_{trk}; #eta; events", {HistType::kTH2F, {{301, -0.5, 300.5}, {35, -4.5, -1.}}}},                                                                //
+      {"Multiplicity", "alibi_nightlies/O2DPG_pp_minbias_testbeam.sh/15-11-2021-18:00 - tf13; collisionID; N_{trk}^{MFT}", {HistType::kTH1F, {{101, 0., 100.}}}, true} //
+    }                                                                                                                                                                  //
+  };
+
+  void process(o2::aod::Collision const& collision, o2::aod::MFTTracks const& tracks)
+  {
+
+    auto z = collision.posZ();
+    registry.fill(HIST("NtrkZvtx"), tracks.size(), z);
+
+    for (auto& track : tracks) {
+      registry.fill(HIST("TracksPhiEta_in_coll"), track.phi(), track.eta());
+      registry.fill(HIST("TracksEtaZvtx"), track.eta(), z);
+      registry.fill(HIST("Multiplicity"), icoll);
+      registry.fill(HIST("NtrkEta"), tracks.size(), track.eta());
+    }
+    icoll++;
+  }
+  //end of process
+
+  //aod::McCollisions
+};
+//end of MyTask
+
+struct analyseGenTracks {
```

Comment:
Any specific reason why this is a separate task and not a an extra process function in a single task?

Reply:
Not really, the only one is that when I tried to put both of the process functions inside a single task, it wouldn't compile, with the following error : 
[WARN] Task analyse-m-f-t-tracks has no inputs
[ERROR] error parsing options of o2-analysis-mm-analyse-mft-tracks: 
unrecognised option '--aod-file'

But it can also be handy when I am processing real data, I can just comment the line adaptAnalysisTask<analyseGenTracks>(cfgc)

Reply:
Each process function, that is not named `process()` needs to have an accompanying `PROCESS_SWITCH()` declaration. See, for example, `multiProcess.cxx` in tutorials.

Reply:
Ok thanks, do you want me to write it all in one task then ?

Reply:
Having it in one task is indeed the preferred way. It also means you would not need to recompile the task to run on MC or data :)

Reply:
Ok, will change that today when I have time